website/frontend/src/tests/pages/Send/Send.test.tsx

```diff
@@ -0,0 +1,497 @@
+    const sendMessageMock = jest.spyOn(ApiService, 'sendMessage').mockResolvedValue();
+
+    const { rerender } = render(
+      <SendMessagePage path="/send/message/:programId" initialEntries={[`/send/message/${PROGRAM_ID}`]} />
+    );
+
+    expect(mockProgram).toBeCalledWith(PROGRAM_ID);
+
+    // header test
+
+    expect(screen.getByText('NFT')).toBeInTheDocument();
+    expect(screen.getByText('New message')).toBeInTheDocument();
+
+    // checking if fields are present in a form
+
+    const valueField = screen.getByLabelText('Value');
+    const payloadField = screen.getByLabelText('Payload');
+    const gasLimitField = screen.getByLabelText('Gas limit');
+    const destinationField = screen.getByLabelText('Destination');
+    const payloadTypeField = screen.getByLabelText('Payload type');
+    const payloadTypeSwitch = screen.getByLabelText('Enter type');
+
+    const sendMessageBtn = screen.getByText('Send message');
+    const calculateGasBtn = screen.getByText('Calculate Gas');
+
+    expect(valueField).toBeInTheDocument();
+    expect(payloadField).toBeInTheDocument();
+    expect(gasLimitField).toBeInTheDocument();
+    expect(destinationField).toBeInTheDocument();
+
+    expect(payloadTypeField).toBeDisabled();
+    expect(payloadTypeField).toBeInTheDocument();
+    expect(payloadTypeField).toHaveValue('Bytes');
+
+    expect(payloadTypeSwitch).toBeInTheDocument();
+
+    expect(sendMessageBtn).toBeInTheDocument();
+    expect(calculateGasBtn).toBeInTheDocument();
+
+    // unauthorized submit
+
+    submit(sendMessageBtn);
+
+    await waitFor(() => expect(screen.getByText('WALLET NOT CONNECTED')).toBeInTheDocument());
+
+    useAccountMock(TEST_ACCOUNT);
+
+    rerender(<SendMessagePage path="/send/message/:programId" initialEntries={[`/send/message/${PROGRAM_ID}`]} />);
+
+    // form validation
+
+    changeFieldValue(valueField, '1000');
+    expect(valueField).toHaveValue(1000);
+
+    changeFieldValue(payloadField, '0x00');
+    expect(payloadField).toHaveValue('0x00');
+
+    changeFieldValue(gasLimitField, '30000000');
+    expect(gasLimitField).toHaveValue('30,000,000');
+
+    changeFieldValue(destinationField, 'program');
+    expect(destinationField).toHaveValue('program');
+
+    fireEvent.click(payloadTypeSwitch);
+
+    expect(payloadTypeField).not.toBeDisabled();
+
+    changeFieldValue(payloadTypeField, 'u32');
+    expect(payloadTypeField).toHaveValue('u32');
+
+    const formValues: FormValues = {
+      value: 1000,
+      payload: '0x00',
+      gasLimit: 30000000,
+      payloadType: 'u32',
+      destination: 'program',
+    };
+
+    // calculate gas
+
+    fireEvent.click(calculateGasBtn);
+
+    expect(calculateGas).toBeCalledTimes(1);
+    expect(calculateGas).toBeCalledWith(
+      'handle',
+      TEST_API,
+      formValues,
+      expect.any(Object),
+      undefined,
+      null,
+      PROGRAM_ID,
+      undefined
+    );
```

Comment:
Don't think that we should expect gas function execution, just check for new alert and field value